### PR TITLE
Allow pushing SDK directly to main

### PR DESCRIPTION
**Pull request by joaopalet**

### CODE REVIEW

File: scripts/sdk-create-pr.sh [hunk 1]
```diff
@@ -6,7 +6,11 @@ ROOT_DIR=$(git rev-parse --show-toplevel)
 COMMIT_NAME="SDK Generator Bot"
 COMMIT_EMAIL="noreply@stackit.de"
 SDK_REPO_LOCAL_PATH="${ROOT_DIR}/sdk-repo-updated" # Comes from generate-sdk.sh
-REPO_BRANCH="main"
```
Comment (joaopalet) [inline scripts/sdk-create-pr.sh:9]: This was unused

File: scripts/sdk-create-pr.sh [hunk 2]
```diff
@@ -6,7 +6,11 @@ ROOT_DIR=$(git rev-parse --show-toplevel)
...
+
+REPO_BRANCH=$1
```
Comment (vicentepinto98) [inline scripts/sdk-create-pr.sh:10]: :+1:

File: scripts/sdk-create-pr.sh [hunk 3]
```diff
@@ -54,11 +58,15 @@ rm -rf ./*
...
-git switch -c "$1"
+if [[ "$REPO_BRANCH" != "main" ]]; then
+    git switch -c "$REPO_BRANCH"
+fi
 git add -A
-if git commit -m "$2"; then # Commit will fail if it doesn't contain any changes
-    git push origin "$1"
-    gh pr create --title "$3" --body "$4" --head "$1" --base "main"
+if git commit -m "$COMMIT_MESSAGE"; then # Commit will fail if it doesn't contain any changes
+    git push origin "$REPO_BRANCH"
```
Comment (vicentepinto98) [inline scripts/sdk-create-pr.sh:66]: But if you are pushing to main, you'll need to add `--force` 
And give permissions to the technical user on ADO to bypass policies when pushing

Comment (joaopalet) [reply scripts/sdk-create-pr.sh:66]: I'll see about the permissions, but why do we need to add `--force`? Isn't it a normal `push`?

Comment (vicentepinto98) [reply scripts/sdk-create-pr.sh:66]: But push to main is not allowed, the branch is protected. So if you don't pass the `--force` flag it will throw error

Comment (joaopalet) [reply scripts/sdk-create-pr.sh:66]: Ah I though the bypassing of policies would allow that already

Comment (joaopalet) [reply scripts/sdk-create-pr.sh:66]: Tested and it worked

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**